frontend/src/components/wallet/DepositDialog.tsx

```diff
@@ -0,0 +1,96 @@
+import { Button } from '../ui/button'
+import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogTrigger } from '../ui/dialog'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import { Input } from '../ui/input'
+import { useDeposit } from '@/hooks/useDeposit'
+import { useTokenContext } from '@/contexts/TokenContext'
+import { useApprove } from '@/hooks/useApprove'
+import { getTokenPreviewByTitle } from '@/utils/getTokenPreviewByTitle'
+import { TokenTitle } from '@/types'
+import { useEffect, useState } from 'react'
+import { DepositTransactionOverview } from './DepositTransactionOverview'
+import { Spinner } from '../custom/Spinner'
+import { useQueryClient } from '@tanstack/react-query'
+import { parseTokenInput } from '@/utils/parseTokenInput'
+
+interface FormType {
+  amount: number | ''
+}
+
+export const DepositDialog = () => {
+  const [open, setOpen] = useState(false)
+  const { selectedToken } = useTokenContext()!
+  const queryClient = useQueryClient()
+  const [underlyingAssetTitle] = useState<TokenTitle>(selectedToken.underlyingAssetTitle!)
+  const approve = useApprove(selectedToken.address)
+  const { deposit, isPending, isTxFinished, isTxLoading } = useDeposit(selectedToken)
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { isValid, isDirty },
+    reset,
+  } = useForm<FormType>({
+    defaultValues: {
+      amount: '',
+    },
+  })
+  const amount = watch('amount')
+  const convertedAmount = parseTokenInput(amount)
+
+  useEffect(() => {
```

Comment:
Instead of useEffect we can leverage tanstack's onSuccess/onError when it comes to transactions